share/cmake/utils/CompilerFlags.cmake

```diff
@@ -40,7 +55,9 @@ elseif(USE_CLANG)
 
     # Use of 'register' specifier must be removed for C++17 support.
     set(PLATFORM_COMPILE_OPTIONS "${PLATFORM_COMPILE_OPTIONS};-Wno-deprecated-register")
-
+    if (HAVE_SSE2 AND HAVE_NEON)
+        set(PLATFORM_COMPILE_FLAGS "${PLATFORM_COMPILE_FLAGS};-march=armv8-a+fp+simd+crypto+crc")
```

Comment:
Should that be `PLATFORM_COMPILE_OPTIONS` now?

Reply:
I read somewhere that those were the default on Apple M1, but I can't find it anymore.
I'm trying to add those back, but I ran into an issue with the universal build since the build for x86_64 doesn't accept those options.

Since there is only one "configure" for the universal build, I'm trying to find a way to set those compiler options based on the architecture at **build time**, but I haven't found a working solution so far.

Do you have any idea @remia? I thought that using generator expression would work, but it does not seems like it does.

Reply:
After some quick search I found [this](https://stackoverflow.com/a/67492623/4814046) but that seem tied to XCode, need to look further. Like you say, may should make sure these flags are needed, maybe bumping the compiler verbose levels or finding the command to show the default flags would help confirming. Also might be worth assessing again why we enable universal binary by default.

Reply:
I've added a comment in my latest commit about this. These flags are needed for ARM platforms that do not enable floating point calculation and/or SIMD instructions by default. On Apple's ARM platform, these features are enabled by default. If they were not enabled by default, we would get a bunch of errors during build about SIMD instructions. (can be tested by adding "nosimd" or "nofp" compiler flags).